Auto-convert spaces to underscores in migration names

lpopedv opened the pull request:
## Summary
Auto-converts spaces to underscores in migration names for `mix ecto.gen.migration`, improving user experience by handling common input patterns automatically.

## Changes
- Modified `Mix.Tasks.Ecto.Gen.Migration` to automatically replace spaces with underscores
- Added `normalize_migration_name/1` function to handle the conversion
- Added comprehensive tests for the new functionality
- Maintained backward compatibility and existing validation

## Motivation
Similar to ORMs like Prisma, this provides better UX by automatically handling spaces in migration names instead of showing a validation error. Users can now run:

```bash
# Before (would error)
mix ecto.gen.migration add posts table

# After (works automatically)
mix ecto.gen.migration add posts table
# Generates: 20240628120000_add_posts_table.exs
```

## Testing
- [x] All existing tests pass
- [x] New tests added for space conversion functionality
- [x] Edge cases tested (multiple spaces, leading/trailing spaces)
- [x] Module name generation verified
- [x] Backward compatibility maintained

josevalim commented:
My concern with this approach is that it now leads to inconsistencies. If a project uses both styles, then someone will point out that they should stick with one, and if we are picking one, then we might as well stick with Elixir’s filename conventions?

lpopedv commented:
> My concern with this approach is that it now leads to inconsistencies. If a project uses both styles, then someone will point out that they should stick with one, and if we are picking one, then we might as well stick with Elixir’s filename conventions?

I understand your concern about consistency. You're right that Elixir follows snake_case conventions!

However, I think the auto-conversion actually **enforces** consistency rather than breaking it - the final result is always snake_case regardless of how the user types it. The generated files will always be consistent.

This is similar to how many tools auto-format code (like `mix format`) - they accept different input styles but produce consistent output.

But I'm happy to discuss alternatives if you have a different preference. What would you suggest as the best approach?

lpopedv commented:
> Apologies, I read the description but not the code, and then I misunderstood the approach being taken.
> 
> there is one issue with your approach is that it assumes all arguments are now part of the migration name. We need to update the tests, because you are testing “add posts table” as argument but, when invoked from the command line, each word is a separate argument. And this may block us from adding further arguments in the future.

Thanks for the feedback, José! No worries at all – you’re right, I hadn’t considered that the CLI passes each word as a separate argument.

I’ve updated the implementation to use [name | rest] and join them before normalization, so mix ecto.gen.migration add posts table now correctly becomes "add_posts_table". I also updated the tests to pass arguments as separate words to match real CLI usage.

Let me know if there’s anything else I should adjust!
